onnx/defs/nn/defs.cc

```diff
@@ -2487,4 +2487,187 @@ ONNX_OPERATOR_SET_SCHEMA(
         )ONNX"
         ));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+  // we need at least two inputs to have a shape for this inference.
+  if (!hasNInputShapes(ctx, 2)) {
+    return;
+  }
+
+  // validating 'input' input
+  auto input_shape = ctx.getInputType(0)->tensor_type().shape();
+  if ((input_shape.dim_size() < 2) || (input_shape.dim_size() > 3)) {
+    fail_shape_inference("input tensor contains partial shape. It must contain either 2 or 3 elements");
+  }
+
+  // validating 'output_shape' input
+  std::vector<int64_t> output_shape;
+  const TensorProto* temp_output_shape = ctx.getInputData(1);
+  if (temp_output_shape) {
+    output_shape = ParseData<int64_t>(temp_output_shape);
+
+    if (output_shape.size() != 2) {
+      fail_shape_inference("output_shape tensor contains partial shape for output. It must contain 2 elements");
```

Comment:
Suggested message "The output_shape tensor is expected to have 2 elements, but it does not." We could also add actual number of elements to make message more useful.